cli/cmd/root.go

```diff
@@ -63,24 +66,38 @@ var rootCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) error {
 		logger.Infof("Running...")
 
-		ctx, cancel := context.WithCancel(cmd.Context())
-		defer cancel()
-
 		cli, err := newCli()
 		if err != nil {
 			return fmt.Errorf("failed to initialize CLI: %w", err)
 		}
 
+		ctx, cancel := context.WithCancel(cmd.Context())
+		defer cancel()
+
+		// Start watching for abort event
+		cli.WatchForAbort(ctx, cancel, DefaultWatcherInterval)
+
+		// Collect non-critical errors for reporting them back to the backend
 		if waitForServerAttached {
 			if err := cli.WaitForVolumeAttachment(ctx); err != nil {
-				return fmt.Errorf("failed to wait for block device being attached: %w", err)
+				err = fmt.Errorf("failed to wait for block device being attached: %w", err)
+				logger.Error(err)
```

Comment:
If we update the local state manager to log the errors passed to MarkDone, then we don't need to remember to log here.

Reply:
Fixed.